### PR TITLE
Add an API compatibility guide for pandas and PySpark users

### PR DESCRIPTION
**Pull request by HyukjinKwon**
Screen Shot 2019-10-10 at 1 26 53 PM
Screen Shot 2019-10-10 at 1 26 58 PM

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
Had an offline discussion. Seems it's not good to lose index column in this example. Let me take a look tomorrow.

**Comment by yifan-cao** [comment]
Looks good. Thanks.

**Comment by HyukjinKwon** [comment]
Let me update this PR soon per #906

File: docs/source/user_guide/options.rst [hunk 1]
```diff
@@ -3,8 +3,6 @@ Options and settings
 ====================
 .. currentmodule:: databricks.koalas
 
-Overview
---------
```
Comment (HyukjinKwon) [inline docs/source/user_guide/options.rst]: Unrelated change but just to make it consistent.

**Comment by HyukjinKwon** [comment]
Let me merge this one.

**Pull request merged**